src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_debug.c

```diff
@@ -1,25 +1,1098 @@
+    GfxPrint_Printf(&printer, "%s", "ｲ");
+    GfxPrint_SetPos(&printer, 4, 26);
+    GfxPrint_Printf(&printer, "%s", "ﾉ");
+    GfxPrint_SetPos(&printer, 4, 27);
+    GfxPrint_Printf(&printer, "%s", "ﾁ");
+
+    // Magic
+    GfxPrint_SetPos(&printer, 8, 25);
+    GfxPrint_Printf(&printer, "%s", "ﾏ");
+    GfxPrint_SetPos(&printer, 8, 26);
+    GfxPrint_Printf(&printer, "%s", "ﾎ");
+    GfxPrint_SetPos(&printer, 8, 27);
+    GfxPrint_Printf(&printer, "%s", "ｳ");
+
+    // Lottery
+    GfxPrint_SetPos(&printer, 13, 25);
+    GfxPrint_Printf(&printer, "%s", "ﾀｶﾗ");
+    GfxPrint_SetPos(&printer, 13, 26);
+    GfxPrint_Printf(&printer, "%s", "ｸｼﾞ");
+
+    // Gold Color (Oceanside Spider House Mask Order)
+    GfxPrint_SetPos(&printer, 23, 24);
+    GfxPrint_Printf(&printer, "%s", "ｷﾝｲﾛ");
+
+    // Bombers (code)
+    GfxPrint_SetPos(&printer, 23, 26);
+    GfxPrint_Printf(&printer, "%s", "ﾎﾞﾝ");
+    GfxPrint_SetPos(&printer, 23, 27);
+    GfxPrint_Printf(&printer, "%s", "ﾊﾞｰｽﾞ");
+
+    *gfxp = GfxPrint_Close(&printer);
+    GfxPrint_Destroy(&printer);
+}
+
+void KaleidoScope_DrawDigit(PlayState* play, s32 digit, s32 rectLeft, s32 rectTop) {
+    OPEN_DISPS(play->state.gfxCtx);
+
+    gDPLoadTextureBlock(POLY_OPA_DISP++, ((u8*)gCounterDigit0Tex + (8 * 16 * digit)), G_IM_FMT_I, G_IM_SIZ_8b, 8, 16, 0,
+                        G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD,
+                        G_TX_NOLOD);
+    gSPTextureRectangle(POLY_OPA_DISP++, rectLeft << 2, rectTop << 2, (rectLeft + 8) << 2, (rectTop + 16) << 2,
+                        G_TX_RENDERTILE, 0, 0, 1 << 10, 1 << 10);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+void KaleidoScope_DrawInventoryEditor(PlayState* play) {
+    Gfx* gfx;
+    Gfx* gfxRef;
+    s32 counterDigits[4];
+    s16 slot;
+    s16 i;
+    s16 j;
+    s32 rectLeft;
+    s32 rectTop;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    func_8012C628(play->state.gfxCtx);
+
+    gDPSetRenderMode(POLY_OPA_DISP++, G_RM_XLU_SURF, G_RM_XLU_SURF2);
+    gDPSetCombineMode(POLY_OPA_DISP++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 0, 0, 0, 220);
+    gDPFillRectangle(POLY_OPA_DISP++, 24, 12, 298, 228);
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetCombineLERP(POLY_OPA_DISP++, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0, 0, 0, 0, PRIMITIVE, TEXEL0, 0,
+                      PRIMITIVE, 0);
+
+    gfxRef = POLY_OPA_DISP;
+    gfx = Graph_GfxPlusOne(gfxRef);
+    gSPDisplayList(OVERLAY_DISP++, gfx);
+
+    KaleidoScope_DrawInventoryEditorText(&gfx);
+
+    gSPEndDisplayList(gfx++);
+    Graph_BranchDlist(gfxRef, gfx);
+    POLY_OPA_DISP = gfx;
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 0, 0, 255);
+    gDPSetEnvColor(POLY_OPA_DISP++, 0, 0, 0, 0);
+
+    // Current Health Quarter (X / 4)
+    KaleidoScope_DrawDigit(play, (((void)0, gSaveContext.save.playerData.health) % 16) / 4, 217, 15);
```

Comment:
```suggestion
    KaleidoScope_DrawDigit(play, (((void)0, gSaveContext.save.playerData.health) % 0x10) / 4, 217, 15);
```
Health is best as hex right?